client/lib/olark/index.js

```diff
@@ -148,10 +156,32 @@ const olark = {
 		olarkEvents.on( 'api.chat.onOperatorsAway', olarkActions.setOperatorsAway );
 		olarkEvents.on( 'api.chat.onOperatorsAvailable', olarkActions.setOperatorsAvailable );
 
+		olarkExpandedEvents.forEach( ( event ) => olarkEvents.on( event, () => {
+			// We query the dom here because there is no other 100% accurate way to figure this out. Olark does not
+			// provide initial events for api.box.onExpand when the api.box.show event is fired.
+			const isOlarkExpanded = !! document.querySelector( '.olrk-state-expanded' );
+			if ( isOlarkExpanded !== olarkStore.get().isOlarkExpanded ) {
+				olarkActions.setExpanded( isOlarkExpanded );
+			}
+		} ) );
+
 		updateDetailsEvents.forEach( event => olarkEvents.on( event, olarkActions.updateDetails ) );
 
 		debug( 'Olark code loaded, beginning configuration' );
 
+		olarkEvents.on( 'api.chat.onCommandFromOperator', ( event ) => {
+			if ( event.command.name === 'end' ) {
+				const { email } = user.get();
```

Comment:
I don't know how `user` operates, but if it's not caching this we might be needlessly getting the data.

Can't say for sure, but I'm guessing we don't change emails during a single session very often